src/book/02-system/03-tools/04-nugan.mdx

```diff
@@ -0,0 +1,214 @@
+
+A [Cycle-Consistent Adversarial Network (CycleGAN)](https://arxiv.org/abs/1703.10593) performs unpaired image-to-image translation. CycleGAN is based on [image-to-image translation GAN (pix2pix)](https://arxiv.org/abs/1611.07004). The main difference between these two GAN models is that CycleGAN uses unpaired data while pix2pix uses paired data. CycleGANs translate some domain A to another domain B, and vice versa. It uses four networks.
+
+| Network         | Description                                            |
+| --------------- | ------------------------------------------------------ |
+| Generator A     | Transfers an image from the A domain to the B domain.  |
+| Generator B     | Transforms an image from the B domain to the A domain. |
+| Discriminator A | Determines if a given image is from the A domain.      |
+| Discriminator B | Determines if a given image is from the B domain.      |
+
+During training, the generators will receive images from the dataset pool for their respective domain. The output of these will be fake images. The fake images are fed into the other generator.
+
+![A flow diagram with two lanes. One lane reads 'Real A' to 'Generator A' to 'Fake B' to 'Generator B' to 'Cycled Fake A'. The other reads 'Real B' to 'Generator B' to 'Fake A' to 'Generator A' to 'Cycled Fake B'.](./images/nugan_cycle.svg 'Cycle Mapping')
+
+The CycleGAN also uses an identity mapping to stabilise the training. It will input a real A image into the B generator, which should give back the original real A input. This is also done for real B images into the A generator.
+
+![A flow diagram with two lanes. One lane reads 'Real B' to 'Generator A' to 'Real B'. The other lane reads 'Real A' to 'Generator B' to 'Real A'.](./images/nugan_identity.svg 'Identity Mapping')
+
+## The Model
+
+This implementation of CycleGAN uses real images and NUpbr paired raw and segmentation images. The NUpbr raw images represent the A domain of the CycleGAN. The real images represent the B domain of the CycleGAN. We train the model on real images and NUpbr images to obtain a dataset of pseudo-real images.
+
+For a given real NUpbr raw image, we will also load in the corresponding real NUpbr segmentation image into the model. We set the non-black (not background) pixels of the segmentation image to white, and ensure the black is true black. This gives us a black and white image. The black portions of the image represent '0' and the white portions of the image represent '1'. This is called our attention mask.
+
+![Segmentation mask and a black and white image where the background of the segmentation mask is black and the rest white.](images/cyclegan_attention.png 'Segmentation image and corresponding attention mask.')
+
+After we get a fake real image, we use the attention mask to add back in the original background.
```

Comment:
what is a fake real image?

Reply:
The output of the generator that tries to map things to realistic images.
Maybe 'fake realistic image'? Or I just define fake/original beforehand.